modules/tensor_mechanics/test/tests/strain_energy_density/tests

```diff
@@ -91,4 +91,14 @@
     requirement = 'The system shall be capable of calculating strain energy rate density when using '
                   'hand-coded Jacobian and weak plane stress.'
   []
+  [tot_elas_stress_name]
+    type = 'CSVDiff'
+    issues = '#25202'
+    input = 'tot_model_stress_name.i'
+    csvdiff = 'tot_model_stress_name_out.csv'
+    rel_err = 1e-6
+    design = 'StrainEnergyDensity.md'
+    requirement = 'The system shall be capable of calculating strain energy density for materials with '
+                  'elastic stress with a user-specified name and small strain.'
```

Comment:
If we provide a customized stress tensor, we are not sure that it is indeed the elastic stress (it should, but cannot guarantee that). I'd remove "elastic stress" from the requirement description.